### ISSUE TITLE
Add Primary Term on-page breadcrumb compatibility

### ISSUE THREAD

**Issue by sybrew** [sybrew/the-seo-framework#407]
Let's put my comment into action: https://github.com/sybrew/the-seo-framework/issues/276#issuecomment-370044202

Make the Primary Term selection affect on-page breadcrumbs.

Candidates:
- WooCommerce
- Breadcrumb NavXT
- ??? _feel free to submit others in a comment below_

**Comment by nlemoine** [comment]
I personally use this library: https://github.com/inc2734/wp-breadcrumbs

It could be a great starting point. One class should handle the crumbs generation returning an object/array you can then use as JSON-LD, display, whatever. 

That's off topic but I stumbled upon this issue wondering why JSON-LD isn't generated on taxonomy page but only on post/page? Any reason this feature is missing?

**Comment by sybrew** [comment]
Thanks for that, @nlemoine! Your comment is better put at #276.

The reason it's not implemented as such is that Google ignores first-level breadcrumbs, as shown in the image below. **Documentation** is the first-level page, whereas **API** is the second-level page.

![image](https://user-images.githubusercontent.com/7662236/58950100-4983f000-878e-11e9-8b3c-f2e6f066f127.png)

This will be changed before I implement #276, as it's a requirement for that.

**Comment by nlemoine** [comment]
Thanks for your answer. However, taxonomies can be hierarchical and one could have more than one term level (eg. Category level 1 > Category level 2 > Category level 3). Would make sense to generate a breadcrumb on taxonomy pages in that case?

**Comment by sybrew** [comment]
Yes. But, back when I implemented it, my main focus was getting it for the posts and pages, not for the terms. 

For now, posts do support your scenario for terms in their breadcrumb output. Hierarchical pages are supported as well.

**Comment by sybrew** [comment]
Punted to the next release. There, we're going to take another look at the structure revolving around this, anyway.

**Comment by sybrew** [comment]
Changed: The last commit removes (almost) all blockers we had. 

We may forgo supporting NavXT since we now generate all structured data and breadcrumbs in one churn -- reducing performance overhead significantly. We only need to add an interface (PHP and user) to control the output.

WooCommerce already had our primary-term support. However, this commit unintentionally adds conflicting output data, which I will rectify before release.

**Comment by sybrew** [comment]
We'll forgo adding bespoke third-party support for now and focus on getting our shortcode ready first. Punted.

**Comment by sybrew** [comment]
Bricks Builder's breadcrumb element might also be a candidate.
